db/refresh.go

```diff
@@ -188,6 +188,66 @@ func (r *refreshTokenRepo) Revoke(userID, token string) error {
 	return tx.Commit()
 }
 
+func (r *refreshTokenRepo) RenewRefreshToken(clientID, userID, oldToken string) (newRefreshToken string, err error) {
```

Comment:
A lot of this logic looks identical to Create and Verify. Can we figure out how to have these share code? E.g. renewing the token should just be "Verify -> Delete -> Create"